server/service/src/processors/transfer/shipment/create_inbound_shipment.rs

```diff
@@ -0,0 +1,158 @@
+            return Ok(None);
+        }
+        // 3.
+        if !matches!(
+            outbound_shipment.invoice_row.status,
+            InvoiceRowStatus::Shipped | InvoiceRowStatus::Picked
+        ) {
+            return Ok(None);
+        }
+        // 4.
+        if linked_shipment.is_some() {
+            return Ok(None);
+        }
+
+        // Execute
+        let new_inbound_shipment =
+            generate_inbound_shipment(connection, &outbound_shipment, record_for_processing)?;
+        let (_, new_inbound_lines) = regenerate_inbound_shipment_lines(
+            connection,
+            &new_inbound_shipment.id,
+            &outbound_shipment,
+        )?;
+
+        InvoiceRowRepository::new(connection).upsert_one(&new_inbound_shipment)?;
+
+        let invoice_line_repository = InvoiceLineRowRepository::new(connection);
+
+        for line in new_inbound_lines.iter() {
+            invoice_line_repository.upsert_one(line)?;
+        }
+
+        let result = format!(
+            "shipment ({}) lines ({:?}) source shipment ({})",
+            new_inbound_shipment.id,
+            new_inbound_lines
+                .into_iter()
+                .map(|r| r.id)
+                .collect::<Vec<String>>(),
+            outbound_shipment.invoice_row.id
+        );
+
+        Ok(Some(result))
+    }
+}
+
+fn generate_inbound_shipment(
+    connection: &StorageConnection,
+    outbound_shipment: &Invoice,
+    record_for_processing: &ShipmentTransferProcessorRecord,
+) -> Result<InvoiceRow, RepositoryError> {
+    let store_id = record_for_processing.other_party_store_id.clone();
+    let name_id = outbound_shipment.store_row.name_id.clone();
+
+    let outbound_shipment_row = &outbound_shipment.invoice_row;
+
+    let status = match &outbound_shipment_row.status {
+        InvoiceRowStatus::Picked => InvoiceRowStatus::Picked,
+        InvoiceRowStatus::Shipped => InvoiceRowStatus::Shipped,
+        _ => InvoiceRowStatus::New,
+    };
+
+    let requisition_id =
+        get_request_requisition_id_from_inbound_shipment(connection, &outbound_shipment_row)?;
+
+    let result = InvoiceRow {
+        id: uuid(),
+        invoice_number: next_number(connection, &NumberRowType::InboundShipment, &store_id)?,
+        r#type: InvoiceRowType::InboundShipment,
+        name_id,
+        store_id,
+        status,
+        requisition_id,
+        name_store_id: Some(outbound_shipment_row.store_id.clone()),
+        their_reference: outbound_shipment_row.their_reference.clone(),
+        // 5.
+        linked_invoice_id: Some(outbound_shipment_row.id.clone()),
+        created_datetime: Utc::now().naive_utc(),
+        picked_datetime: outbound_shipment_row.picked_datetime,
+        shipped_datetime: outbound_shipment_row.shipped_datetime,
+        transport_reference: outbound_shipment_row.transport_reference.clone(),
+        // Default
+        colour: None,
+        user_id: None,
+        comment: None,
```

Comment:
do we not create a comment in mSupply? I would have expected the usual 'from requisition #' and any original comment being appended

Reply:
I think you might have updated this in develop to make the comment. I am wondering if the comment is editable by the user, should we change slightly from how it's done in mSupply and display that information based on available graphql (with links to that requisition etc ?)